flask/src/main/groovy/net/corda/gradle/flask/FlaskPlugin.groovy

```diff
@@ -0,0 +1,58 @@
+package net.corda.gradle.flask
+
+import org.gradle.api.GradleException
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.plugins.JavaApplication
+import org.gradle.api.plugins.JavaPlugin
+import org.gradle.api.plugins.JavaPluginConvention
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Copy
+import org.gradle.api.tasks.JavaExec
+import org.gradle.api.tasks.SourceSet
+import org.gradle.jvm.tasks.Jar
+
+class FlaskPlugin implements Plugin<Project> {
+
+    @Override
+    void apply(Project project) {
+        project.getPluginManager().apply(JavaPlugin.class);
+        JavaPluginConvention javaPluginConvention = project.convention.getPlugin(JavaPluginConvention.class)
+        Provider<SourceSet> flaskSourceSetProvider = javaPluginConvention.sourceSets.register("flask")
+        Provider<Copy> extractLauncherTarProvider = project.tasks.register("extractLauncherTar", Copy) {
+            into(project.layout.buildDirectory.dir("classes/flask-launcher"))
+            from(project.tarTree(LauncherResource.instance))
+        }
+
+        project.dependencies {
+            add(flaskSourceSetProvider.get().compileOnlyConfigurationName, extractLauncherTarProvider.get().outputs.files)
+        }
```

Comment:
this tightly couples the `FlaskJarTask` to the `flask` source set, making it impossible to create and successfully use other instances of the task, I'd like to leave users with the ability to create their own flask jars (maybe without even applying the plugin, just adding it to the classpath)

Reply:
Hmm, I've done considerably less here than you seem to think. None of the directories, configurations or dependencies has changed, although I have added the `build/classes/flask-launcher` directory to the "flask" source set.

If you don't want to do that, then you're going to have to replace
```java
flaskSourceSet.output.dir([builtBy: extractLauncherTarProvider], flaskDir)
```
with something like
```java
project.tasks.named(flaskSourceSet.compileJavaTaskName, JavaCompile) {
    dependsOn extractLauncherTarProvider
}
```
because otherwise I don't see anything to tell Gradle that the launcher's tar file must be unpacked before trying to compile the "flask" source set.

Reply:
what I mean is that I don't want the extracted launcher to be included in the source set output, it must be added by the FlaskJar task and here it should only be included in the compile classpath, so that you can always create custom a FlaskJar that works correctly (inject the launcher code in the archive)

Reply:
I managed to find a solution replacing 
```
flaskSourceSet.output.dir([builtBy: extractLauncherTarProvider], flaskDir)
```
with
```
flaskSourceSet.compiledBy(extractLauncherTarProvider)
```
but I'm also getting a flock of warnings like
```
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/home/r3/.gradle/wrapper/dists/gradle-5.6.4-all/9x9myuz2g40on4481pe9i1f1m/gradle-5.6.4/lib/groovy-all-1.3-2.5.4.jar) to field java.util.jar.JarFile.hasClassPathAttribute
```
I know they could be easily suppressed, but I wasn't receiving them before and I'd like to understand what triggered them

Reply:
I think that warning means that this block in the root `build.gradle` doesn't contain an `--add-opens` for the `java.util.jar` package:
```groovy
if (current().java9Compatible) {
    jvmArgs '--add-opens', 'java.base/java.io=ALL-UNNAMED',
            '--add-opens', 'java.base/java.lang=ALL-UNNAMED',
            '--add-opens', 'java.base/java.lang.invoke=ALL-UNNAMED',
            '--add-opens', 'java.base/java.util=ALL-UNNAMED',
            '--add-opens', 'java.base/sun.nio.fs=ALL-UNNAMED',
            '--illegal-access=warn'
}
```
Happily, this block will go away when Gradle 6.7.1 compiles using the Java 8 toolchain.

Reply:
I think `flaskSourceSet.compiledBy` is OK: Gradle _should_ now realise that `compileFlaskJava` has a "compileOnly" dependency on the `flask-launcher` directory, and that the `flask-launcher` directory is built by the `extractLauncherTar` task :+1:.

Reply:
it turns out that the reason of all those Illegal reflective access is the import of `java.util.jar.JarFile` to be able to write
```
exclude("META-INF/MANIFEST.MF")
```
since that archive it as tar file file whose manifest is actually included by accident by the explosion of `slf4j-simple` (or any of its dependencies), I decided to simply remove the manifest from the tar file and solve the problem at the root